app/src/main/java/com/tananaev/passportreader/MainActivity.java

```diff
@@ -318,8 +340,12 @@ private ReadTask(IsoDep isoDep, BACKeySpec bacKey) {
 
         private DG1File dg1File;
         private DG2File dg2File;
+        private DG14File dg14File;
+        private SODFile sodFile;
         private String imageBase64;
         private Bitmap bitmap;
+        private boolean chipAuthSucceeded = false;
```

Comment:
Why do you need class variables? Please make them local.

Reply:
I use class variables here because I need to retrieve the result in onPostExecute. I also put dg14Encoded in the class variables because I need it in the two new functions doPassiveAuth and doChipAuth